idunn/geocoder/nlu_client.py

```diff
@@ -212,27 +226,43 @@ async def get_category():
 
         if category:
             return Intention(
+                type=IntentionType.CATEGORY,
                 filter={"category": category, "bbox": bbox},
                 description={"category": category, "place": place},
             )
 
         return Intention(
-            filter={"q": cat_query, "bbox": bbox}, description={"query": cat_query, "place": place}
+            type=IntentionType.BRAND,
+            filter={"q": cat_query, "bbox": bbox},
+            description={"query": cat_query, "place": place},
         )
 
     async def build_intention_category(self, cat_query, is_brand=False):
         category = await self.classify_category(cat_query, is_brand)
 
         if category:
-            return Intention(filter={"category": category}, description={"category": category})
+            return Intention(
+                type=IntentionType.CATEGORY,
+                filter={"category": category},
+                description={"category": category},
+            )
 
-        return Intention(filter={"q": cat_query}, description={"query": cat_query})
+        return Intention(
+            type=IntentionType.BRAND,
+            filter={"q": cat_query},
+            description={"query": cat_query},
+        )
 
     @classmethod
     def is_poi_request(cls, tags_list):
         """Check if a request is addressed to a POI"""
         return any(t.get("tag") in NLU_POI_TAGS for t in tags_list)
 
+    @classmethod
+    def is_street_request(cls, tags_list):
+        """Check if a request is addressed to a POI"""
+        return all(t.get("tag") in NLU_STREET_TAGS for t in tags_list)
```

Comment:
What is the expected behavior here?  
On one hand, if any query related to a street or an address should match, then I think that other tags such as "city" should be allowed too.
On the other hand, similarly to `is_poi_request` we could also check that any "street" tag is present, using `any` instead of `all` to catch all queries where an address part could disturb a direct search.